lib/common/types/result.go

```diff
@@ -53,8 +54,10 @@ func (r *Result) Decode(reader io.Reader) (*Result, error) {
 
 	for {
 		b, err := common.ReadByte(reader)
-		if err != nil {
+		if errors.Is(err, io.EOF) {
```

Comment:
@noot about that one, if there is no data left in the reader, should we error out or just break and leave data empty? :thinking:

Reply:
i think we can just break if there's no data left